src/Internal/Payment/State/InitialState.php

```diff
@@ -45,10 +81,32 @@ public function __construct(
 	 * @throws Order_Not_Found_Exception Order could not be found.
 	 */
 	public function process() {
-		$order_id       = $this->get_context()->get_order_id();
-		$payment_method = $this->get_context()->get_payment_method();
-		$this->order_service->set_payment_method_id( $order_id, $payment_method->get_id() );
+		$context  = $this->get_context();
+		$order_id = $context->get_order_id();
+
+		// Start by setting up all local objects.
+		$this->order_service->import_order_data_to_payment_context( $order_id, $context );
+		$context->set_metadata( $this->order_service->get_payment_metadata( $order_id ) );
+		$context->set_level3_data( $this->level3_service->get_data_from_order( $order_id ) );
+
+		// Customer management involves a remote call.
+		$customer_id = $this->customer_service->get_or_create_customer_id_from_order(
+			$context->get_user_id(),
+			$this->order_service->get_order( $order_id )
+		);
+		$context->set_customer_id( $customer_id );
+
+		// Payments are currently based on intents, request one from the API.
+		try {
+			$intent = $this->payment_request_service->create_intent( $context );
+		} catch ( Invalid_Request_Parameter_Exception | Extend_Request_Exception | Immutable_Parameter_Exception $e ) {
+			return $this->create_state( SystemErrorState::class );
+		}
+
+		// Intent available, complete processing.
+		$this->order_service->update_order_from_successful_intent( $order_id, $intent, $context );
```

Comment:
maybe we could rename this function to `update_order_from_successful_intent( $order_id, $payment_object, $contect)`

Reply:
This is the same as the definition, I think you forgot to change the thing you had in mind.

Reply:
I think it is related to make function abstract which you addressed in previous comments. You can ignore the original comment.